elements/lisk-chain/test/unit/chain.spec.ts

```diff
@@ -271,6 +271,56 @@ describe('chain', () => {
 		});
 	});
 
+	describe('newStateStore', () => {
+		beforeEach(async () => {
+			chainInstance['_lastBlock'] = newBlock({ height: 532 });
+			stubs.dependencies.storage.entities.Block.get.mockResolvedValue([
+				newBlock(),
+				genesisBlock,
+			]);
+		});
+
+		it('should populate the chain state with genesis block', async () => {
+			chainInstance['_lastBlock'] = newBlock({ height: 1 });
+			await chainInstance.newStateStore();
+			await expect(
+				stubs.dependencies.storage.entities.Block.get,
+			).toHaveBeenCalledWith(
+				{
+					height_gte: 1,
+					height_lte: 1,
+				},
+				{ limit: null, sort: 'height:desc' },
+			);
+		});
+
+		it('should return with the chain state from last block to last block - 309', async () => {
```

Comment:
This statement is a bit confusing, maybe `should return the chain state from lastBlock.height to lastBlock.height - 309`